TestStack.Dossier.Tests/BuilderBuildTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Shouldly;
+using TestStack.Dossier.Tests.Builders;
+using TestStack.Dossier.Tests.Entities;
+using Xunit;
+
+namespace TestStack.Dossier.Tests
+{
+    public class BuilderBuildTests
+    {
+        [Fact]
+        public void GivenBasicBuilder_WhenCallingBuildExplicitly_ThenReturnAnObject()
+        {
+            var builder = Builder<Customer>.CreateNew();
+
+            var customer = builder.Build();
+
+            customer.ShouldBeOfType<Customer>();
+        }
+
+        [Fact]
+        public void GivenBuilder_WhenCallingSetExplicitly_ShouldOverrideValues()
+        {
+            var builder = Builder<Customer>.CreateNew()
```

Comment:
Have to agree with you it's surprising! I've had to add a check for property.CanWrite now (because it was trying to write to a property with just a getter) and it returns true for all of the properties with private set and property.SetValue is able to write to the property.